File: lab3_1/lib/ferma.dart
import 'dart:math';

ferma(int n) {
  if (n < 0) return 'must be > 0';
  if (n % 2 == 0) return 'is not odd';
  int x = sqrt(n).toInt();
  int y = 0;
  while (true) {
    int r = (x * x) - n;
    if (r > 0) {
      y = sqrt(r).toInt();
    }
    if ((y * y) == r) {
      int a = x - y;
      int b = x + y;
      return '$a * $b = $n';
    }
    x++;
  }
}

File: lab3-js/js/ferma.js
const ferma = n => {
	if (isNaN(n)) return { error: 'not a number' }
	if (n < 0) return { error: 'must be > 0' }
	if (!(n & 1)) return { error: 'is not odd' }
	let x = parseInt(Math.sqrt(n))
	let y = 0
	while (true) {
		let r = x * x - n
		console.log(r)
		y = parseInt(Math.sqrt(r))
		console.log(y)
		if (y * y === r) {
			return { result: [x - y, x + y] }
		}
		x++
	}
}

console.log(ferma(13))
